lib/instrumentation/modules/aws-sdk/sns.js

```diff
@@ -0,0 +1,102 @@
+const constants = require('../../../constants')
+
+const TYPE = 'messaging'
+const SUBTYPE = 'sns'
+const ACTION = 'PUBLISH'
+const ACCESS_POINT = 'accesspoint'
+const PHONE_NUMBER = '<PHONE_NUMBER>'
+
+function getRegionFromRequest (request) {
+  return request && request.service &&
+        request.service.config && request.service.config.region
+}
+
+function getSpanNameFromRequest (request) {
+  const topicName = getDestinationNameFromRequest(request)
+  return `SNS ${ACTION} ${topicName}`
+}
+
+function getMessageContextFromRequest (request) {
+  return {
+    queue_name: getDestinationNameFromRequest(request)
+  }
+}
+
+function getMessageDestinationContextFromRequest (request) {
+  return {
+    resource: `${SUBTYPE}/${getDestinationNameFromRequest(request)}`,
+    type: TYPE,
+    name: SUBTYPE,
+    cloud: {
+      region: getRegionFromRequest(request)
+    }
+  }
+}
+
+function getDestinationNameFromRequest (request) {
+  const phoneNumber = request && request.params && request.params.PhoneNumber
+  if (phoneNumber) {
+    return PHONE_NUMBER
+  }
+
+  let arn = request && request.params && request.params.TopicArn
+  if (!arn) {
+    arn = request && request.params && request.params.TargetArn
+  }
+
+  if (!arn) {
+    return
+  }
+  const parts = arn.split(':')
+  const topicOrAccessPointName = parts.pop()
+  const maybeAccessPoint = parts.pop()
+  if (ACCESS_POINT === maybeAccessPoint) {
+    return `${ACCESS_POINT}:${topicOrAccessPointName}`
+  }
+  return topicOrAccessPointName
+}
+
+function instrumentationSns (orig, origArguments, request, AWS, agent, { version, enabled }) {
+  const type = TYPE
+  const subtype = SUBTYPE
+  const action = ACTION
+  const operation = request.operation
+  if (operation !== 'publish') {
+    return orig.apply(request, origArguments)
+  }
```

Comment:
Should handle `ignore_message_queues` per https://github.com/elastic/apm/blob/master/specs/agents/tracing-instrumentation-messaging.md#ignore_message_queues-configuration

Reply:
Ah, right, good call.  I've added some code to ignore instrumentation for certain queues.